python/GafferUI/AnimationEditor.py

```diff
@@ -313,6 +316,52 @@ def __sourceCurvePlug( self, plug ) :
 		assert( isinstance( result, Gaffer.Animation.CurvePlug ) )
 		return result
 
+	def __animationGadgetContextMenu( self, *unused ) :
+
+		import IECore
+		import functools
+
+		# convert mouse position to event line
+		line = self.__gadgetWidget.getViewportGadget().rasterToGadgetSpace(
+			imath.V2f( GafferUI.Widget.mousePosition( relativeTo = self.__gadgetWidget ) ), gadget = self.__animationGadget )
+
+		# only show context menu when mouse not above axis
+		if self.__animationGadget.onTimeAxis( line ) or self.__animationGadget.onValueAxis( line ) :
+			return False
+
+		# check there are selected keys
+		emptySelectedKeys = not self.__animationGadget.selectedKeys()
+
+		# key interpolation for selected keys
+		interpolation = None if emptySelectedKeys else self.__curveEditor.keyWidget().getInterpolationForSelectedKeys()
+
+		# build context menu
+		menuDefinition = IECore.MenuDefinition()
+
+		for mode in sorted( Gaffer.Animation.Interpolation.values.values() ) :
+			menuDefinition.append(
+				"/Set Interpolation/%s" % ( mode.name ),
+				{
+					"command" : functools.partial(
+						Gaffer.WeakMethod( self.__setSelectedKeysInterpolation ),
+						mode=mode
+					),
+					"active" : not emptySelectedKeys,
```

Comment:
This is another spot where we should be considering `Metadata::readOnly()`. I suppose one super simple way of dealing with this is just to call `self.setEnabled( False )` for the entire editor?